Library/Homebrew/requirements.rb

```diff
@@ -125,16 +125,28 @@ class JavaDependency < Requirement
 
   satisfy { java_version }
 
+  env do
+    java_home = Pathname.new(@java_home)
+    ENV["JAVA_HOME"] = java_home
+    ENV.prepend_path "PATH", java_home/"bin"
+    if (java_home/"include").exist? # Oracle JVM
+      ENV.append_to_cflags "-I#{java_home}/include"
+      ENV.append_to_cflags "-I#{java_home}/include/darwin"
+    else # Apple JVM
+      ENV.append_to_cflags "-I/System/Library/Frameworks/JavaVM.framework/Versions/Current/Headers/"
+    end
+  end
+
   def initialize(tags)
-    # ignore trailing +
-    @version = tags.shift.sub(/\+$/, "") if /(\d\.)+\d/ === tags.first
+    @version = tags.shift if /(\d\.)+\d/ === tags.first
```

Comment:
Ok.

However I quick scanned all the Requirements. The behaviors are inconsistent. [Some](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/requirements/maximum_macos_requirement.rb) doesn't remove the tags from superclass. [Other](https://github.com/Homebrew/homebrew/blob/master/Library/Homebrew/requirements/language_module_dependency.rb) deliberately passes the arguments to superclass. And many of them who call `shift` fail to redefine `inspect`.